plasmapy/diagnostics/tests/test_langmuir.py

```diff
@@ -60,50 +57,47 @@ def test_fit_func_double_lin_inverse(self):
         expect_value = 8
 
         errStr = (
-            f"Linear fitting function with inverse slope and an offset for"
-            f"use in fitting a bi-Maxwellian electron current growth"
-            f"region did not return the expected value {expect_value} and"
-            f"instead returned {value}"
+            f"Linear fitting function with inverse slope and an offset for use in fitting a bi-Maxwellian electron current growth region did not return the expected value {expect_value} and instead returned {value}"
         )
         assert value == expect_value, errStr
 
 
 class Test__characteristic_errors:
-    r"""Test the Characteristic class constructor in langmuir.py"""
+    """Test the Characteristic class constructor in langmuir.py"""
 
-    bias_2darr = np.array((np.random.rand(N), np.random.rand(N))) * u.V
+    bias_2darr = np.array((np.random.rand(N), np.random.rand(N))) * u.A
     current_2darr = np.array((np.random.rand(N), np.random.rand(N))) * u.A
 
-    bias_infarr = np.append(np.random.rand(N - 1), np.inf) * u.V
+    bias_infarr = np.append(np.random.rand(N - 1), np.inf) * u.A * u.V
     current_infarr = np.append(np.random.rand(N - 1), np.inf) * u.A
 
-    bias_longarr = np.random.rand(N + 1) * u.V
+    bias_longarr = np.random.rand(N + 1) * u.A * u.V
     current_longarr = np.random.rand(N + 1) * u.A
 
-    current_arr2 = np.random.rand(N) * u.A
+    current_arr2 = np.random.rand(N) * u.A *u.V
```

Comment:
Could you say a bit more about why the units are being changed?  I'm wondering if there would need to be a bugfix in the corresponding code.